### ISSUE TITLE
Vamp Gateway Agent 094rc throws error on kubernetes

### ISSUE THREAD

**Issue by olafmol** [magneticio/vamp-docker-images#29]
running on minikube with k8s 1.5 deploying vamp-gateway-agent:katana using the manual steps gives this error:

```
2017-03-21T07:42:20.708116603Z VAMP GATEWAY AGENT: katana [0.9.3-26-g684a11b]
2017-03-21T07:42:20.712510941Z confd 0.11.0
2017-03-21T07:42:20.713331629Z VAMP_KEY_VALUE_STORE_TYPE       : etcd
2017-03-21T07:42:20.713437733Z VAMP_KEY_VALUE_STORE_CONNECTION : http://etcd-client:2379
2017-03-21T07:42:20.713510014Z VAMP_KEY_VALUE_STORE_PATH       : /vamp/gateways/haproxy/1.7/configuration/value
2017-03-21T07:42:20.713593459Z VAMP_ELASTICSEARCH_URL          : http://elasticsearch:9200
2017-03-21T07:42:20.717839248Z creating confd configuration and template
2017-03-21T07:42:21.710924326Z dnsmasq/run: disabling
2017-03-21T07:42:22.131290989Z ok: down: dnsmasq: 1s, normally up
2017-03-21T07:42:22.131556949Z reaped child process 54 (0)
2017-03-21T07:42:25.734556126Z 2017-03-21T07:42:25Z vamp-gateway-agent-1lgw7 /usr/bin/confd[69]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:42:30.762980752Z 2017-03-21T07:42:30Z vamp-gateway-agent-1lgw7 /usr/bin/confd[104]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:42:35.803548294Z 2017-03-21T07:42:35Z vamp-gateway-agent-1lgw7 /usr/bin/confd[121]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:42:40.868345197Z 2017-03-21T07:42:40Z vamp-gateway-agent-1lgw7 /usr/bin/confd[137]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:42:45.889096574Z 2017-03-21T07:42:45Z vamp-gateway-agent-1lgw7 /usr/bin/confd[154]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:42:50.905813443Z 2017-03-21T07:42:50Z vamp-gateway-agent-1lgw7 /usr/bin/confd[171]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:42:55.943379457Z 2017-03-21T07:42:55Z vamp-gateway-agent-1lgw7 /usr/bin/confd[188]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:43:00.981963952Z 2017-03-21T07:43:00Z vamp-gateway-agent-1lgw7 /usr/bin/confd[205]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:43:06.017152126Z 2017-03-21T07:43:06Z vamp-gateway-agent-1lgw7 /usr/bin/confd[222]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:43:11.038668293Z 2017-03-21T07:43:11Z vamp-gateway-agent-1lgw7 /usr/bin/confd[239]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:43:16.056441409Z 2017-03-21T07:43:16Z vamp-gateway-agent-1lgw7 /usr/bin/confd[256]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:43:21.074013406Z 2017-03-21T07:43:21Z vamp-gateway-agent-1lgw7 /usr/bin/confd[272]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:43:26.088689219Z 2017-03-21T07:43:26Z vamp-gateway-agent-1lgw7 /usr/bin/confd[289]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:43:31.108132086Z 2017-03-21T07:43:31Z vamp-gateway-agent-1lgw7 /usr/bin/confd[306]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:43:36.124757372Z 2017-03-21T07:43:36Z vamp-gateway-agent-1lgw7 /usr/bin/confd[322]: ERROR 100: Key not found (/vamp) [7]
2017-03-21T07:43:41.150582569Z 2017-03-21T07:43:41Z vamp-gateway-agent-1lgw7 /usr/bin/confd[338]: ERROR 100: Key not found (/vamp) [7]
```

**Comment by olafmol** [comment]
deploying the 093 version of VGA gives the same error btw. Tested with the entire 093 package worked fine. This makes me suspect that the issue comes from the Vamp 094/katana docker package/config.

**Comment by ItsMeijers** [comment]
This happens due to VAMP not running yet. vamp-gateway-agent recovers once vamp runs and the key has been created. In the future it would be nice to have vamp running before we start vga, but will close this for now.
